internal/gatewayapi/securitypolicy.go

```diff
@@ -105,65 +111,126 @@ func (t *Translator) ProcessSecurityPolicies(securityPolicies []*egv1a1.Security
 					if _, ok := gatewayRouteMap[gw]; !ok {
 						gatewayRouteMap[gw] = make(sets.Set[string])
 					}
-					gatewayRouteMap[gw].Insert(utils.NamespacedName(route).String())
+					gatewayRouteMap[gw].Insert(utils.NamespacedName(targetedRoute).String())
+					parentGateways = append(parentGateways, gwv1a2.ParentReference{
+						Group:       ptr.To(gwapiv1.Group(gwapiv1.GroupName)),
+						Kind:        ptr.To(gwapiv1.Kind(KindGateway)),
+						Namespace:   ptr.To(gwapiv1.Namespace(namespace)),
+						Name:        p.Name,
+						SectionName: p.SectionName,
+					})
 				}
 			}
 
-			err := t.translateSecurityPolicyForRoute(policy, route, resources, xdsIR)
 			if err != nil {
-				status.SetSecurityPolicyCondition(policy,
+				status.SetSecurityPolicyCondition(
```

Comment:
hi @zhaohuabing, can this be merged after #2846, since it introduces a more generic method for `PolicyStatus`?

Reply:
Sure, I'll change this PR to draft and revisit it after 2846.